### ISSUE TITLE
DeepLab training on own dataset Issue Assign Shape Error

### ISSUE THREAD

**Issue by karanchahal** [tensorflow/models#6166]
I am also training on my custom dataset. It has just 2 classes foreground and background and I am using the mobile net pretrained weights. However I am receiving this error:
```
Restoring from checkpoint failed. This is most likely due to a mismatch between the current graph and the graph from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:

Assign requires shapes of both tensors to match. lhs shape= [2] rhs shape= [19]
```
I am using the following command:
```
%cd /content/models/research

# From tensorflow/models/research/
!python deeplab/train.py \
    --logtostderr \
    --training_number_of_steps=30000 \
    --train_split="train" \
    --model_variant="mobilenet_v2" \
    --output_stride=16 \
    --decoder_output_stride=4 \
    --train_crop_size=513 \
    --train_crop_size=513 \
    --train_batch_size=12 \
    --dataset="bdd" \
    --initialize_last_layer = True \
    --fine_tune_batch_norm = False \
    --tf_initial_checkpoint=deeplab/deeplabv3_mnv2_cityscapes_train/model.ckpt\
    --train_logdir=deeplab/graphs/ \
    --dataset_dir=deeplab/bdd_seg/tfrecord
```

My dataset description looks like this:
```


_DATASETS_INFORMATION = {
    'cityscapes': _CITYSCAPES_INFORMATION,
    'pascal_voc_seg': _PASCAL_VOC_SEG_INFORMATION,
    'ade20k': _ADE20K_INFORMATION,
    'bdd': _BDD_INFORMATION,
}

```

I am quite confused. Why is this error happening ?

**Comment by ilous12** [comment]
Hi, @karanchahal 

Did you check to have any assets in your exp directory? You must remove all exp folder.
try again.

**Comment by sercant** [comment]
I think the problem is that you set `initialize_last_layer=True` which should be `False` since you are changing the number of classes in the final prediction.

**Comment by kayvonkhosrowpour** [comment]
Hi @karanchahal,
Were you able to find a solution? I am having the same issue.

**Comment by karanchahal** [comment]
Couldn't get to work :/ If you guys are interested in instance segmentation the Mateeport Mask RCNN repo is pretty good. Also check out papers with code, they have great implementations.

**Comment by kayvonkhosrowpour** [comment]
> 
> 
> Couldn't get to work :/ If you guys are interested in instance segmentation the Mateeport Mask RCNN repo is pretty good. Also check out papers with code, they have great implementations.

Can you recommend a purely semantic segmentation repo? My dataset is not instance-based.

**Comment by iris0329** [comment]
Seems like same problem.
I try to fine tune the model on my dataset(about 3200images and only one label), I choose mobileNet v2 as backnone but seems `mobilenet_v2_1.3_224` and `mobilenet_v2_1.4_224` doesn't work because **Tensor shape doesn't mismatch.** 
Only `mobilenet_v2_1.0_224` can be used to fine- tune the model (you can have a try), but the accuracy is not enough.
But I really want to use the newest mobilenet v2.

**Comment by kayvonkhosrowpour** [comment]
Edit:
Resolved issue: https://github.com/tensorflow/models/issues/6559#issuecomment-482671438

**Comment by Angus-Lee** [comment]
> Hi, @karanchahal
> 
> Did you check to have any assets in your exp directory? You must remove all exp folder.
> try again.

Indeed

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be an old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
